gateway/gateway-runtime/policy-engine/internal/admin/server_test.go

```
@@ -356,3 +402,292 @@ func TestIPWhitelistMiddleware_PreservesRequestPath(t *testing.T) {
+	// A client capped at TLS 1.1 cannot complete the handshake against a
+	// listener whose floor is TLS 1.2.
+	httpsClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true, //nolint:gosec // test-only self-signed cert
+				MinVersion:         tls.VersionTLS10,
+				MaxVersion:         tls.VersionTLS11,
+			},
+		},
+	}
+	_, err := httpsClient.Get(fmt.Sprintf("https://127.0.0.1:%d/health", tlsPort))
```

golangci-lint (2.12.2)

[error] response body must be closed

(bodyclose)